packages/account-management/src/accounts/strategies/account-rpc-strategy.ts

```diff
@@ -0,0 +1,208 @@
+/**
+ * Direct Chain Indexer Strategy
+ * Queries the chain directly to find existing smart accounts
+ * This is a fallback when indexers are unavailable
+ *
+ * How it works:
+ * 1. Calculate predicted instantiate2 address from authenticator (using crypto utilities)
+ * 2. Check if contract exists at that address via RPC
+ * 3. If exists, query contract state for authenticators
+ * 4. Return account info
+ */
+
+import { CosmWasmClient } from "@cosmjs/cosmwasm-stargate";
+import {
+  calculateSalt,
+  calculateSmartAccountAddress,
+  AUTHENTICATOR_TYPE,
+  type AuthenticatorType,
+} from "@burnt-labs/signers";
+import type {
+  IndexerStrategy,
+  SmartAccountWithCodeId,
+} from "../../types/indexer";
+import { Buffer } from "buffer";
+
+export interface RpcAccountStrategyConfig {
+  /** RPC URL for querying the chain */
+  rpcUrl: string;
+  /** Contract checksum (hex) for instantiate2 calculation */
+  checksum: string;
+  /** Creator/fee granter address */
+  creator: string;
+  /** Address prefix (e.g., "xion") */
+  prefix: string;
+  /** Code ID of the smart account contract */
+  codeId: number;
+}
+
+/**
+ * Direct Chain Indexer Strategy
+ * Calculates predicted address and queries chain directly (no indexer needed)
```

Comment:
We should add a note that this does not account for meta accounts where the authenticator in question has been removed or added to other accounts.